### PR TITLE
[16.0][MIG] mrp_warehouse_calendar

### PR DESCRIPTION
**Pull request by JoanSForgeFlow**
Standard Migration to v16

@ForgeFlow

### CODE REVIEW

File: mrp_warehouse_calendar/models/mrp_production.py [hunk 1]
```diff
@@ -8,8 +8,8 @@ class MrpProduction(models.Model):
     _inherit = "mrp.production"
 
     @api.onchange("date_planned_start", "product_id")
-    def _onchange_date_planned_start(self):
-        res = super(MrpProduction, self)._onchange_date_planned_start()
+    def _compute_date_planned_finished(self):
+        res = super(MrpProduction, self)._compute_date_planned_finished()
```
Comment (LoisRForgeFlow) [inline mrp_warehouse_calendar/models/mrp_production.py]: be carefull with the decorator, does the compute have/need an api.onchange?

File: mrp_warehouse_calendar/models/mrp_production.py [hunk 2]
```diff
@@ -7,9 +7,9 @@
 class MrpProduction(models.Model):
     _inherit = "mrp.production"
 
-    @api.onchange("date_planned_start", "product_id")
-    def _onchange_date_planned_start(self):
-        res = super(MrpProduction, self)._onchange_date_planned_start()
+    @api.depends("company_id", "date_planned_start", "is_planned", "product_id")
+    def _compute_date_planned_finished(self):
+        res = super(MrpProduction, self)._compute_date_planned_finished()
         if self.date_planned_start and not self.is_planned:
```
Comment (BernatPForgeFlow) [inline mrp_warehouse_calendar/models/mrp_production.py]: Use a loop "for production in self" to not get singleton errors.

**Review by BernatPForgeFlow — APPROVED** [review]
Code and functional review OK!

File: mrp_warehouse_calendar/models/mrp_production.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from odoo import api, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    @api.depends("company_id", "date_planned_start", "is_planned", "product_id")
+    def _compute_date_planned_finished(self):
+        for production in self:
+            res = super(MrpProduction, self)._compute_date_planned_finished()
```
Comment (LoisRForgeFlow) [inline mrp_warehouse_calendar/models/mrp_production.py]: you are creating a loop inside a loop here by calling super with all records inside the for.

you should call super first outside the loop, then iterate (ideally a filtered recordset) and update what's needed.

**Review by LoisRForgeFlow — APPROVED** [review]

**Pull request merged**